### PR TITLE
Add search catalog and document-types endpoints

### PR DESCRIPTION
**Pull request by cummings**
Add GET /search/catalog
Add GET /document_types

### CODE REVIEW

File: test/spec/api/document-types.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var documentTypes = require('../../../lib/api/document-types');
+var MIME_TYPES = require('../../../lib/mime-types');
+
+// fixtures
+var apiOptions = {
+    baseUrl: 'https://api.mendeley.com',
+    authFlow: function() {},
```
Comment (customcommander) [inline test/spec/api/document-types.spec.js]: This `,` doesn't seem to be needed.

File: test/spec/api/search.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var search = require('../../../lib/api/search');
+var MIME_TYPES = require('../../../lib/mime-types');
+
+// fixtures
+var apiOptions = {
+    baseUrl: 'https://api.mendeley.com',
+    authFlow: function() {},
```
Comment (customcommander) [inline test/spec/api/search.spec.js]: Redundant `,` I think

File: test/spec/api/document-types.spec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    method: 'GET',
+                    resource: '/document_types',
+                    headers: { 'Accept': MIME_TYPES.DOCUMENT_TYPE }
+                })
+            );
+        });
+
+        it('returns api object with retrieve property', function() {
+            var documentTypesApiMethods = Object.keys(documentTypes(apiOptions));
+            expect(documentTypesApiMethods).toContain('retrieve');
```
Comment (customcommander) [inline test/spec/api/document-types.spec.js]: Would it make more sense to test that `retrieve` is a function instead? If I read this correctly, if `retrieve` is set to an array, the test will pass, yet the code would be wrong.

File: test/spec/api/search.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                    method: 'GET',
+                    resource: '/search/catalog',
+                    headers: { 'Accept': MIME_TYPES.DOCUMENT }
+                })
+            );
+        });
+
+        it('returns api object with catalog search property', function() {
+            var searchApiMethods = Object.keys(search(apiOptions));
+            expect(searchApiMethods).toContain('catalog');
```
Comment (customcommander) [inline test/spec/api/search.spec.js]: Same comment as above.

File: lib/api/document-types.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var utils = require('../utilities');
+var MIME_TYPES = require('../mime-types');
+
+/**
+ * Search API
+ *
+ * @namespace
+ * @name api.search
```
Comment (customcommander) [inline lib/api/document-types.js]: Does this comment need update? ;)

File: test/spec/api/document-types.spec.js (hunk 1 above)
Comment (lazlojuly) [inline test/spec/api/document-types.spec.js]: image

File: test/spec/api/document-types.spec.js (hunk 3 above)
Comment (lazlojuly) [inline test/spec/api/document-types.spec.js]: @customcommande, Good idea to remove `toContain`. There is a better way: 

```
const api = documentTypes(apiOptions);
expect(api).to.have.property('retrieve').and.to.be.a('function')
```

Did not run it but the exact syntax should be very close.

Comment (lazlojuly) [inline test/spec/api/document-types.spec.js]: Ignore this comment. I forgot Jasmine is used for testing here (not mocha-chai...) I will update this with Jasmine assertion accordingly.

**Pull request closed without merging**